spirv_msl.cpp

```diff
@@ -6596,6 +6596,12 @@ string CompilerMSL::to_function_args(VariableID img, const SPIRType &imgtype, bo
 		else if (sampling_type_needs_f32_conversion(coord_type))
 			tex_coords = convert_to_f32(tex_coords, 1);
 
+		if (msl_options.texture_1D_as_2D)
+		{
+			string base_type = is_fetch ? "uint" : "float";
+			tex_coords = base_type + "2(" + tex_coords + ", 0)";
```

Comment:
Needs to be y = 0.5 for float UV to avoid potential issues with sampling on edges.

Reply:
Good catch! Fixed now.